mountaineer/client_builder/builder.py

```diff
@@ -480,18 +462,81 @@ async def build_javascript_chunks(self, max_concurrency: int = 25):
 
         # Go through the exceptions, logging the build errors explicitly
         has_build_error = False
+        final_exception: str = ""
         for result in results:
             if isinstance(result, Exception):
                 has_build_error = True
                 if isinstance(result, BuildProcessException):
                     secho(f"Build error: {result}", fg="red")
-                else:
-                    raise result
+                final_exception += str(result)
 
         if has_build_error:
-            raise BuildProcessException(
-                "Build process failed. Errors are listed in the console."
-            )
+            raise BuildProcessException(final_exception)
+
+        self.move_build_artifacts_into_project()
+
+    def move_build_artifacts_into_project(self):
```

Comment:
This is better practice in general to ensure our build process doesn't leave local cache files in a partially-valid state. It was specifically necessary for this PR because we want to use the prebuilt ExceptionController SSR files in the case of a larger build exception. With our previous logic, we would clear out the Exception SSR file and might skip recreating it if the build terminates early.